.github/workflows/cache-cleanup.yaml

```diff
@@ -5,12 +5,15 @@ on:
   pull_request:
     types:
       - closed
+      - labeled
 
 jobs:
   cleanup:
     runs-on: ubuntu-latest
     steps:
       - name: Cleanup
+        # when the PR is closed or labeled as 'Stale'
+        if: ${{ github.event.action == 'closed' || github.event.label.name == 'Stale' }}
```

Comment:
does the case of the strings here matter? you have lowercase `closed` and Titlecase `Stale`...

Reply:
if the event is being generated as "Stale" then let's jusst use that.